[HACK][Merchant AI key] Add feature flag & render settings row if eligible

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->


<!-- Id number of the GitHub issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

<!-- 
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
🛠 Need an AI review?  
Add a comment: `@coderabbitai review`
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
-->

## Description
As part (1) of cleaning up and submitting [this HACK week project](https://github.com/woocommerce/woocommerce-ios/pull/15372), this PR adds a feature flag and renders the AI settings row under the app's Menu.

## Testing information
- Switch `allowMerchantAIAPIKey` to `true` 
- Navigate to Menu, observe there's an `AI Settings` row: 
- Tapping on it will render an empty view, and log the `hub_menu_ai_settings_tapped` event:

```
🔵 Tracked hub_menu_ai_settings_tapped, properties: [store_id: c5bd46cc-1804-4f7b-badb-bb98c449127f, was_ecommerce_trial: false, plan: , blog_id: -1, is_wpcom_store: false, site_url: https://indiemelon.mystagingwebsite.com]
```

![Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 (US site) - 2025-03-24 at 15 01 33](https://github.com/user-attachments/assets/8911e766-8de0-45ed-9fa8-41feaa6df4ea)



---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
Add allowMerchantAIAPIKey feature flag

iamgabrielma committed:
Render AI Settings row if eligible. Log tap event

iamgabrielma committed:
switch flag off

dangermattic commented:
<!--
  1 Error: Please ensure the PR has the <...
  0 Warnings
  2 Messages: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please ensure the PR has the <code>category: tracks</code> label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
localization

iamgabrielma committed:
switch flag off (again!)

jaclync reviewed (approved):
LGTM 🚀 maybe the HACK Week p2 can be linked in the PR or the linked main GitHub issue?

> Switch allowMerchantAIAPIKey to true

How about setting the feature flag to return `buildConfig == .localDeveloper || buildConfig == .alpha` for easier developement & PR testing?

> Tapping on it will render an empty view, and log the hub_menu_ai_settings_tapped event

There's already an event `hub_menu_option_tapped` with `option: ai` property logged by default. WDYT about just using this event, with the `option` value maybe more specific like `ai-settings`?

iamgabrielma committed:
Merge branch 'trunk' into hack/allowMerchantAIAPIKey-feature-flag

# Conflicts:
#	WooCommerce/WooCommerce.xcodeproj/project.pbxproj

iamgabrielma committed:
make row visible on dev config

iamgabrielma committed:
track tap event as existing with better property description

iamgabrielma commented:
> LGTM 🚀 maybe the HACK Week p2 can be linked in the PR or the linked main GitHub issue?

Done!

> How about setting the feature flag to return buildConfig == .localDeveloper || buildConfig == .alpha for easier developement & PR testing?

Yeah, I was wary of this because is an empty view for now 😅 . Updated here: https://github.com/woocommerce/woocommerce-ios/pull/15404/commits/780b9452f21aa324c4dce2513564687f7adb373b

> There's already an event hub_menu_option_tapped with option: ai property logged by default. WDYT about just using this event, with the option value maybe more specific like ai-settings?

Oh, better! Updated: https://github.com/woocommerce/woocommerce-ios/pull/15404/commits/89f9f4a9c31008a87ec3e9eee241f864ef8f8956

iamgabrielma committed:
Add flag to mock flag service and fix failing tests
